scripts/globals/events/harvest_festivals.lua

```diff
@@ -184,58 +190,58 @@ function onHalloweenTrade(player, trade, npc)
                     local ghost = 368
                     local hound = 365
                     local skeleton = 564
+                    local gob = math.random(484, 511)
+                    local gigas = math.random(707, 711)
+                    local demon = math.random(740, 756)
                     local darkStalker = math.random(531, 534)
 
-                    local halloweenCostumeList = { quadav, orc, yagudo, shade, ghost, hound, skeleton, darkStalker }
+                    local halloweenCostumeList = { quadav, orc, yagudo, shade, ghost, hound, skeleton, darkStalker, gob, gigas, demon }
 
                     local costumePicked = halloweenCostumeList[math.random(1, #halloweenCostumeList)] -- will randomly pick one of the costumes in the list
                     player:addStatusEffect(xi.effect.COSTUME, costumePicked, 0, 3600)
-
-                    -- pitchForkCostumeList defines the special costumes per zone that can trigger the pitch fork requirement
-                    -- zone, costumeID
-                    local pitchForkCostumeList =
-                    {
-                        234, shade, skeleton, -- Bastok mines
-                        235, hound, ghost,    -- Bastok Markets
-                        230, ghost, skeleton, -- Southern Sandoria
-                        231, hound, skeleton, -- Northern Sandoria
-                        241, ghost, shade,    -- Windurst Woods
-                        238, shade, hound     -- Windurst Woods
-                    }
-
-                    for zi = 1, #pitchForkCostumeList, 3 do
-                        if
-                            zone == pitchForkCostumeList[zi] and
-                            (
-                                costumePicked == pitchForkCostumeList[zi + 1] or
-                                zone == pitchForkCostumeList[zi] and
-                                costumePicked == pitchForkCostumeList[zi + 2]
-                            )
-                        then -- Gives special hint for pitch fork costume
-                            player:messageSpecial(ID.text.IF_YOU_WEAR_THIS)
-
-                        elseif zi == 16 then
-                            player:messageSpecial(ID.text.THANK_YOU_TREAT)
-                        end
-                    end
+                    player:messageSpecial(ID.text.TRICK_OR_TREAT)
                 else
                     player:messageSpecial(ID.text.THANK_YOU)
                 end
 
-                if not alreadyTradedChk then
-                    player:setCharVar(varName, utils.mask.setBit(harvestFestTreats, itemInList, true))
-                end
-
                 player:tradeComplete()
+            end
+        end
+    end
+end
+
+xi.events.harvest.applyHalloweenDecorations = function(zoneid)
+    if xi.events.harvest.isHalloweenEnabled() ~= 0 then
+        local decoration = zones[zoneid].npc.HALLOWEEN_DECORATIONS
```

Comment:
We don't have these defined in our ID.lua files. Moreover we need to set deco:setAlwaysRender(true) on static decorations. Else they only pop in when you're 40 yalms near them.

Reply:
Also have to take ID shifts into account here.